runtime/compiler/env/VMJ9.cpp

```diff
@@ -8429,15 +8429,8 @@ TR_J9SharedCacheVM::isClassVisible(TR_OpaqueClassBlock * sourceClass, TR_OpaqueC
 
    if (comp->getOption(TR_UseSymbolValidationManager))
       {
-      TR::SymbolValidationManager *svm = comp->getSymbolValidationManager();
-      SVM_ASSERT_ALREADY_VALIDATED(svm, sourceClass);
-      SVM_ASSERT_ALREADY_VALIDATED(svm, destClass);
-      validated = true;
-      }
-   else
-      {
-      validated = ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) sourceClass) &&
-                  ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) destClass);
+      bool validated = comp->getSymbolValidationManager()->addIsClassVisibleRecord(sourceClass, destClass, validated);
+      SVM_ASSERT(validated, "Failed to validate addIsClassVisibleRecord");
```

Comment:
This isn't correct as it stands for a few reasons:
* You should leave the `else` path as is, because the SVM isn't always enabled
* You're passing in `validated` to `addIsClassVisibleRecord` but that's incorrect because
  1. `validated` is always `false` at this point
  2. `validated` is what `addIsClassVisibleRecord` returns; you want to pass in the answer that `addIsClassVisibleRecord` returns.
  3. You shouldn't `SVM_ASSERT` here; it's ok for a front end query to fail to add a validation record, in which case we just won't use that information.

You should do something like
```
bool
TR_J9SharedCacheVM::isClassVisible(TR_OpaqueClassBlock * sourceClass, TR_OpaqueClassBlock * destClass)
   {
   TR::Compilation* comp = _compInfoPT->getCompilation();
   TR_ASSERT(comp, "Should be called only within a compilation");

   bool validated = false;
   bool isVisible = TR_J9VMBase::isClassVisible(sourceClass, destClass);

   if (comp->getOption(TR_UseSymbolValidationManager))
      {
      validated = comp->getSymbolValidationManager()->addIsClassVisibleRecord(sourceClass, destClass, isVisible);
      }
   else
      {
      validated = ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) sourceClass) &&
                  ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) destClass);
      }

   return (validated ? isVisible : false);
   }
```

Reply:
Fixed in [d6c066e](https://github.com/eclipse-openj9/openj9/pull/16000/commits/d6c066e8bde78201876f47a5d510e2f2fc733b49).